Lamina/include/GL/camera.hpp

```diff
@@ -0,0 +1,145 @@
+#ifndef LM_CAMERA
+#define LM_CAMERA
+
+#include "../Lamina/include/Math/vector.hpp"
+#include "../Lamina/include/GL/window.hpp"
+#include "../Lamina/include/Math/constants.hpp"
+#include "../Lamina/include/user_interface/event.hpp"
+
+namespace lm
+{
+	static lm::mat4 LookAtMatrix(vec3f position, vec3f target, vec3f up)
+	{
+		lm::vec3f direction = lm::normalize(position - target);
+		lm::vec3f right = lm::normalize(lm::CrossProduct(direction, up));
+		lm::mat4 first({
+			{right.x(),		right.y(),		right.z(),		0},
+			{up.x(),		up.y(),			up.z(),			0},
+			{direction.x(),	direction.y(),	direction.z(),	0},
+			{0,				0,				0,				1}
+			});
+		lm::mat4 second({
+			{1, 0, 0, -position.x()},
+			{0, 1, 0, -position.y()},
+			{0, 0, 1, -position.z()},
+			{0, 0, 0, 1}
+			});
+		return first * second;
+	}
+
+	class PivotCamera
+	{
+	public:
+		PivotCamera() : target(lm::vec3f({ 0, 0, 0 })), position(lm::vec3f({ 0, 0, 0 })), radius(5), pitch(0), yaw(0), window(nullptr), pitchClamp(89.f), 
```

Comment:
The lines in this file are too long and are inconvenient for people with small displays or when trying to work with a split screen. If you don't want to format it manually, you can use an automatic code formatter which will automatically limit line length to something more reasonable.